packages/cosmos_ui_components/lib/components/cosmos_text_field.dart

```diff
@@ -42,17 +49,33 @@ class _CosmosTextFieldState extends State<CosmosTextField> {
       controller: controller,
       maxLines: widget.maxLines,
       maxLength: widget.maxLength,
-      onChanged: widget.onChanged,
+      keyboardType: widget.keyboardType,
+      onChanged: (value) {
+        widget.onChanged(value);
+        setState(() {});
```

Comment:
this will cause a lot of rebuilds on each character change. could you optimize the logic and call the `setState` only if the text changes from empty to nonempty state and vice versa?

Reply:
Updated